app/controllers/hyrax/batch_ingest/batches_controller.rb

```diff
@@ -59,10 +59,11 @@ def show
       private
 
         def available_admin_sets
-          # TODO: Restrict available_admin_sets to only those current user has access to.
-          @available_admin_sets ||= AdminSet.all.map do |admin_set|
-            [admin_set.title.first, admin_set.id]
+          # Restrict available_admin_sets to only those current user can desposit to.
+          @available_admin_sets ||= Hyrax::Collections::PermissionsService.admin_set_ids_for_user(Hyrax::PermissionTemplateAccess::DEPOSIT, current_ability).map do |admin_set_id|
+            [AdminSet.find(admin_set_id).title.first, admin_set_id]
```

Comment:
It would be good to do a solr query here to get the titles instead of a fedora `find`.  There are probably some examples of this in hyrax code that probably involve using a search builder...actually it looks like `Hyrax::AdminSetSearchBuilder` does this and the current work you're doing here:
https://github.com/samvera/hyrax/blob/master/app/search_builders/hyrax/admin_set_search_builder.rb#L38-L44

Reply:
admin_set_ids_for_deposit is what i used at first but i got Ruby error as it's a private method. As for solr query to get admin-set title, I think i know what you mean, if you have an example at hand that would be great; otherwise i'll do some search...

Reply:
alternatively, I have the commented code just below that part, it looks a bit cleaner, but I don't know if it's less efficient.